synapse/util/caches/treecache.py

```diff
@@ -57,4 +60,33 @@ def pop(self, key, default=None):
                 break
             node_and_keys[i+1][0].pop(k)
 
+        popped, cnt = _strip_and_count_entires(popped)
+        self.size -= cnt
         return popped
```

Comment:
I'd quite like to keep the `pop` method returning a single item when given a full key, as that matches expectations.